Prox/Prox/Widgets/ExpandingCardView.swift

```diff
@@ -0,0 +1,135 @@
+public class ExpandingCardView: UIView {
+
+    // MARK: public API
+
+    public override var backgroundColor: UIColor? {
+        get { return scrollView.backgroundColor }
+        set { scrollView.backgroundColor = newValue }
+    }
+    public var cornerRadius: CGFloat {
+        get { return scrollView.layer.cornerRadius }
+        set { scrollView.layer.cornerRadius = newValue }
+    }
+
+    public var shadowColor: CGColor? {
+        get { return layer.shadowColor }
+        set { layer.shadowColor = newValue }
+    }
+    public var shadowOpacity: Float {
+        get { return layer.shadowOpacity }
+        set { layer.shadowOpacity = newValue }
+    }
+    public var shadowRadius: CGFloat {
+        get { return layer.shadowRadius }
+        set { layer.shadowRadius = newValue }
+    }
+    public var shadowOffset: CGSize {
+        get { return layer.shadowOffset }
+        set { layer.shadowOffset = newValue }
+    }
+
+    public var contentView: UIView? {
+        didSet {
+            oldValue?.removeFromSuperview()
+            for c in contentViewConstraints { c.isActive = false }
+            contentViewConstraints.removeAll(keepingCapacity: true)
+
+            guard let contentView = contentView else { return }
+            for v in [contentView] + contentView.subviews { // TODO: okay to disable caller autoresize masks?
+                v.translatesAutoresizingMaskIntoConstraints = false
+            }
+            self.scrollView.addSubview(contentView)
+            contentViewConstraints += tlbrConstraintsEqual(scrollView, contentView)
+
+            // Constrain the width so we don't scroll horizontally.
+            contentViewConstraints.append(scrollView.widthAnchor.constraint(equalTo: contentView.widthAnchor))
+            NSLayoutConstraint.activate(contentViewConstraints)
+        }
+    }
+
+    public init() {
+        super.init(frame: .zero)
+        initStyle()
+        initLayout()
+    }
+
+    public required init?(coder aDecoder: NSCoder) { fatalError("init(coder:) has not been implemented") }
+
+
+    // MARK: implementation.
+
+    private lazy var heightConstraint: NSLayoutConstraint = {
+        // Default constant is high enough to see (for debugging purposes) but low enough
+        // that the caller knows they've done something wrong.
+        let constraint = self.heightAnchor.constraint(equalToConstant: 1)
+
+        // Slightly less than required so a caller doesn't accidentally override
+        // this constraint they don't know about.
+        constraint.priority = UILayoutPriorityRequired - 1
+        return constraint
+    }()
+
+    /// A sequence of constraints from the current user-defined contentView.
+    private var contentViewConstraints: [NSLayoutConstraint] = []
+
+    private lazy var scrollView: UIScrollView = {
+        let view = ObservableScrollView()
+        view.onContentSizeUpdated = { [unowned self] contentSize in
```

Comment:
Note that this is typically done use the delegate pattern (`ObservableScrollViewDelegate` with `didUpdateContentSize`, for example). That way, the client code isn't responsible for breaking the ref cycle, which is error prone.